### ISSUE TITLE
Ugly font on Windows 10

### ISSUE THREAD

**Issue by wdscxsj** [sqlitebrowser/sqlitebrowser#2744]
On Windows 10 Simplified Chinese Edition, the old-style, unpleasant SimSun font is used by sqlitebrowser as the main UI font almost everywhere, even though the OS default font is Microsoft Yahei (微软雅黑).

![Screenshot118](https://user-images.githubusercontent.com/863369/122174589-6b53f980-ceb5-11eb-8552-a9ad5d9bd0bb.png)

Could we have Microsoft Yahei (微软雅黑) back, or an option to select one's favorite? Thanks.

- OS: Windows 10 20H2 Simplified Chinese Edition.
- sqlitebrowser: 3.12.0, 3.12.2. It seems that 3.12.2 uses more SimSun than 3.12.0.

**Comment by mgrojo** [comment]
This might be due to saved settings from a previous version. What happens if you use the Restore Defaults button?

**Comment by wdscxsj** [comment]
> This might be due to saved settings from a previous version. What happens if you use the Restore Defaults button?

Thanks for your suggestion! I tried "Restore Default" on v3.12.0 and v3.12.2, but in both cases the default UI font was still SimSun.

Please note in my previous screenshot, there is a field typeset in Microsoft Yahei. I guess that's not controlled by a setting.

**Comment by chrisjlocke** [comment]
> Please note in my previous screenshot, there is a field typeset in Microsoft Yahei.

This is an example of what the data browser font looks like, which is controlled by the top font setting

![image](https://user-images.githubusercontent.com/7872218/122721887-0dcefc80-d269-11eb-9cdb-d2fd2616a52a.png)

**Comment by wdscxsj** [comment]
@chrisjlocke Sorry I was not quite clear on this point. The Font field is set to 微软雅黑 (aka Microsoft Yahei), but on this dialog, this setting only affects the field containing "Preview only (N/A)". Other fields are still in SimSun.

The Font field may not be designed to control the dialog font, but this doesn't actually matter. The desirable behavior is to inherit the OS default UI font, or have an option to set it manually.

**Comment by chrisjlocke** [comment]
Completely agree - was just explaining what that field was for, and why that one field was in the correct font.  😁

**Comment by mgrojo** [comment]
In theory, we are getting the default font for the application according to [documentation](https://doc.qt.io/qt-5/qapplication.html#font), and then we set the configured size in `General > Font size`, so it might be a problem in Qt.

**Comment by wdscxsj** [comment]
@mgrojo You are right, thanks! It turns out there are plenty of complaints about the ugly default font of Qt on a Chinese Edition of Windows. Could we add a dialog box or something for a lovely user to set her favorite?

Not a Qt developer, I come up with this tiny program in PyQt5:

```python
import sys
from PyQt5.QtWidgets import QApplication, QMainWindow, QPushButton
from PyQt5.QtGui import QFont


class MainWindow(QMainWindow):
    def __init__(self):
        super().__init__()
        self.setWindowTitle("My App 我的应用")
        button = QPushButton("Press Me 点我")
        self.setCentralWidget(button)


def main():
    app = QApplication(sys.argv)

    # Allow a chance to set default GUI font.
    font = QFont()
    font.setFamily('Microsoft Yahei')
    app.setFont(font)

    window = MainWindow()
    window.show()
    app.exec_()


if __name__ == '__main__':
    sys.exit(main())
```

Without the line `app.setFont(font)`, the default font is SimSun:

![1](https://user-images.githubusercontent.com/863369/122851804-30811400-d342-11eb-9dae-06bc7c1b3ced.png)

With `app.setFont(font)`, we can use Microsoft Yahei or anything:

![2](https://user-images.githubusercontent.com/863369/122851977-7211bf00-d342-11eb-90e0-4f82e2dc0e86.png)

**Comment by mgrojo** [comment]
Ideally, Qt should fix the bug, and we update to the new version, otherwise we could end up with too many options which will confuse users. The font for the application should be configured at the desktop level, and not application per application. Haven't you found a reference to a bug report which we can monitor from this issue?

**Comment by wdscxsj** [comment]
Hi, it's reported back in 2011 as [QTBUG-19309](https://bugreports.qt.io/browse/QTBUG-19309) "Wrong default font on Windows 7". The conclusion was SimSun had to be the fallback font due to support for Windows XP.

It's reported again in 2020 as [QTBUG-87391](https://bugreports.qt.io/browse/QTBUG-87391) "Default font in Chinese Windows 10 is wrong". It seems that Qt6 has fixed this issue but not yet Qt5.

The ugly default SimSun was called by bosbyj the "SimSun Ghost in QT" in this [Zeal issue](https://github.com/zealdocs/zeal/issues/536) and I totally agree. This [StackOverflow question](https://stackoverflow.com/questions/7043904/microsoft-sans-serif-font-not-used-in-chinese-localized-windows) was also fun to read. But as a PC user from the last century, I'm quite sure that Chinese users are NOT used to that look & feel.

By the way, it seems that this behavior cannot be overridden using the Windows font link mechanism (`HKLM\SOFTWARE\Microsoft\Windows NT\CurrentVersion\FontLink\SystemLink`).

**Comment by FearYuzu** [comment]
In Japanese Windows Environment, it's more ugly fonts at default.
The only saddest things in this useful app :/
I hope to fix this as fast as possible, being an all UI font customizable is better if possible.

Just installed the latest version and didn't tweak any locate setting in Windows.

![ss_20210722](https://user-images.githubusercontent.com/17077920/126646647-b2017ba6-7afa-4bfe-9e78-c66d9d3654ae.png)

**Comment by justinclift** [comment]
Hmmm, it's been about a year since this bug report.  Maybe we should figure out a way to check for the presence of the "Microsoft Yahei" font (on Windows), and if it's present we use that as the default.

**Comment by wdscxsj** [comment]
I think the default "modern" UI font differs on Chinese, Japanese, and Korean Windows systems. My personal recommendation is to follow the practice of EmEditor (which uses WTL instead Qt, but faced a similar issue for the dialog font), as described in its [v21.3 release announcement](https://www.emeditor.com/emeditor-core/emeditor-v21-3-0-released-including-technical-review/), Nov 2021:

> We have decided to hard-code Microsoft YaHei for Simplified Chinese, and Microsoft JhengHei for Traditional Chinese. We have not changed the other language UI yet but will test the other languages in beta versions for the next version. We will use Meiryo UI for Japanese, Malgun Gothic for Korean, and Segoe UI for other languages.

The "will use" part was done in [EmEditor v21.4](https://www.emeditor.com/text-editor-features/history/new-in-version-21-4/), Dec 2021. And no change ever since.

By the way, this change to EmEditor was proposed by me, "another customer using the Chinese UI" mentioned in the release announcement. As Yukata (author of the EmEditor) is from Japan, we can be sure that the new look is good on at least Simplified Chinese Windows and Japanese Windows.

**Comment by yusen2024** [comment]
Would it be possible to use **Segoe UI Variable Display** as the UI font?
It has excellent rendering quality, is also the default UI font in the Chinese version of **Windows 11**, and offers very good readability while also looking visually pleasing.

**Comment by yusen2024** [comment]
A similar issue and its solution can be found at: https://github.com/FreeCAD/FreeCAD/issues/20471

Additionally, Google’s AI-generated summary for the search query “qt windows ui font” is as follows:

For a Qt application on Windows, **the default UI font usually matches the system's Segoe UI**, but Qt sometimes defaults to an older method, requiring you to explicitly set **QFont("Segoe UI")** for proper modern appearance, especially on HiDPI displays, or use a generic font family like **QFont("Sans Serif")** for better cross-platform fallback. To ensure consistent, native-like Windows look and feel, explicitly setting the font to "Segoe UI" within your Qt application is often the best approach. 
### How to Set the Font in Qt
* For the Entire Application:
``` cpp
#include <QApplication>
#include <QFont>

int main(int argc, char *argv[]) {
    QApplication app(argc, argv);
    QFont font("Segoe UI"); // Or "Segoe UI Semibold", "Segoe UI Light", etc.
    font.setPixelSize(11); // Adjust size as needed
    app.setFont(font);
    // ... your application code ...
    return app.exec();
}
```
This sets "Segoe UI" as the default for all widgets, mimicking the Windows standard.

* For Specific Widgets/Styles:
You can use **qApp->setFont(QFont("QMenu"))** to get system-specific fonts for particular elements like menus, or use **QFont("Sans Serif")** for generic families that systems map well.
* For HiDPI/Modern Displays:
Explicitly setting the font to **QFont("Segoe UI")** with a pixel size helps ensure it renders correctly on high-resolution displays, as older Qt methods might fail. 
---
### Key Font Considerations
* **Segoe UI**: The standard, clean, humanist sans-serif font for Windows (including Windows 10/11), available in various weights.
* **Cross-Platform**: For apps intended for Linux/macOS too, use **QFont("Sans Serif")** or **QFont("Roboto")** as a fallback, or bundle and vendor fonts in a **.qrc** file for perfect consistency.
